api/extensions/ext_celery.py

```diff
@@ -37,34 +37,74 @@ def __call__(self, *args: object, **kwargs: object) -> object:
     }
 
     celery_app.conf.update(
-        result_backend=app.config.get("CELERY_RESULT_BACKEND"),
+        result_backend=dify_config.CELERY_RESULT_BACKEND,
         broker_transport_options=broker_transport_options,
         broker_connection_retry_on_startup=True,
     )
 
-    if app.config.get("BROKER_USE_SSL"):
+    if dify_config.BROKER_USE_SSL:
         celery_app.conf.update(
             broker_use_ssl=ssl_options,  # Add the SSL options to the broker configuration
         )
 
     celery_app.set_default()
     app.extensions["celery"] = celery_app
 
+    schedule = get_schedule(app.config)
+
     imports = [
         "schedule.clean_embedding_cache_task",
         "schedule.clean_unused_datasets_task",
     ]
-    day = app.config.get("CELERY_BEAT_SCHEDULER_TIME")
+
     beat_schedule = {
         "clean_embedding_cache_task": {
             "task": "schedule.clean_embedding_cache_task.clean_embedding_cache_task",
-            "schedule": timedelta(days=day),
+            "schedule": schedule,
         },
         "clean_unused_datasets_task": {
             "task": "schedule.clean_unused_datasets_task.clean_unused_datasets_task",
-            "schedule": timedelta(days=day),
+            "schedule": schedule,
         },
     }
     celery_app.conf.update(beat_schedule=beat_schedule, imports=imports)
 
     return celery_app
+
+
+def get_schedule(app_config):
+    """Determine the schedule type based on configuration."""
+
+    # Fetch configuration values
+    scheduler_type = dify_config.CELERY_BEAT_SCHEDULER_TYPE  # Options: 'time' or 'cron'
+    day = dify_config.CELERY_BEAT_SCHEDULER_TIME
+    cron_expression = dify_config.CELERY_BEAT_SCHEDULER_CRON_EXPRESSION
+
+    # Determine the schedule based on the 'scheduler_type' value using match-case
+    match scheduler_type:
+        case "cron":
+            if not cron_expression:
+                raise ValueError(
+                    "Configuration Error: 'CELERY_BEAT_SCHEDULER_CRON_EXPRESSION' is required when using the 'cron' scheduler type.")
+            try:
+                minute, hour, day_of_month, month_of_year, day_of_week = cron_expression.split()
```

Comment:
How about use [cron-validator](https://pypi.org/project/cron-validator/) to check and parse the units rather than just spliting the string in spaces?